### ISSUE TITLE
Order Tagging

### ISSUE THREAD

**Issue by cj4c0b1** [ccxt/ccxt#7009]
I'm already implemented and using on my custom python version tagged orders and now i want to move this to the ccxt standard library as an Enchantement suggestion.

Order tagging is already supported by a lot of exchanges but is not treated inside ccxt standards.

So i believe that a simple approach to maintain a TAG field in the orders structure will solve this.

In my solution I'm controlling the tag when the order was created, than I maintain this tag control for each order and it can be used for a lot of estrategies and logging.

orders structure suggestion :

{
    'id':                '12345-67890:09876/54321', // string
*   'Clorid':                '12345-67890:09876/54321', // string
*   'Tag':                'trap_54321', // string
    'datetime':          '2017-08-17 12:42:48.000', // ISO8601 datetime of 'timestamp' with milliseconds
    'timestamp':          1502962946216, // order placing/opening Unix timestamp in milliseconds
    'lastTradeTimestamp': 1502962956216, // Unix timestamp of the most recent trade on this order
    'status':     'open',         // 'open', 'closed', 'canceled'
    'symbol':     'ETH/BTC',      // symbol
    'type':       'limit',        // 'market', 'limit'
    'side':       'buy',          // 'buy', 'sell'
    'price':       0.06917684,    // float price in quote currency
    'amount':      1.5,           // ordered amount of base currency
    'filled':      1.1,           // filled amount of base currency
    'remaining':   0.4,           // remaining amount to fill
    'cost':        0.076094524,   // 'filled' * 'price' (filling price used where available)
    'trades':    [ ... ],         // a list of order trades/executions
    'fee': {                      // fee info, if available
        'currency': 'BTC',        // which currency the fee is (usually quote)
        'cost': 0.0009,           // the fee amount in that currency
        'rate': 0.002,            // the fee rate (if available)
    },
    'info': { ... },              // the original unparsed order structure as is
}

The same idea for Clorid. *

* The main reason is because some exchanges do not have tagging and some relay on Clorid.
Actually CCXT translate Clorid as ID, but do not manage appropriately the id and Clorid.

and because of some particularities of some information and exchange erros, I started to create and maintain more structures off the exchange, like balances and order control.

The "problem" is that ccxt in some way have the basic structures but in the overall number of implemented exchanges do not maintain this data structures accurately.

So, the idea here is to start with a basic control on the ccxt side for tagging, even for exchanges that do not support tagging. And in the same way extend the functionalities for other controls, already in place but not 100% accurately done.

This can be extended to :
- balance
- orders (tagging,..)
- open orders (maintain open orders) # for large numbers of orders that can impact order by order cycles
- canceled orders (maintain open orders) #  for large numbers of orders that can impact order by order cycles

best regars @kroitor

**Comment by kroitor** [comment]
Hey @cj4c0b1 ! 

First of, thanks for the feedback, really appreciate it! 

As you've noticed, we're already on the path towards clientOrderId unification, though, this is still a work in progress. We have recently added a bunch of edits related to clientOrderId – it just wasn't documented in the Manual yet.

I've also uploaded a few edits to the Manual to reflect the recent changes (sometimes the Manual may be behind the actual codebase, because this lib is a work in progress, as you know).

> Order tagging is already supported by a lot of exchanges but is not treated inside ccxt standards.

Can you plz list out the specific exchanges that support tagging at this time?

**Comment by cj4c0b1** [comment]
Hej @kroitor 

Good to see that progress. I saw the commits yesterday regarding the ClorId.

About the order tagging I will be more specific in the way some exchanges support tagging.
We can consider "tagging" as a not explicit part of ClorId but some solutions provide a layer over the exchange API to group orders and support that functionality as ccxt can also do.
So, we can consider this in 4 flavors spread in the wild API ecosystems.
1 - explicit tag as part of the order structure, (some alternative platforms as a layer over the exchange API like CTS, alertatron, globitex, etc... ) that can be incorporated in ccxt.
2 - ClorId can be prefixed or postfixed as a tag. (Bitmex, Bitfinex, FTX, Binance, almost all that has ClorId as a string), this is a implicit method.
3 - do not has ClorId. 
4 - No ClorId support and no order tagging

So my first approach was to use the ClorId as a place to prefix or suffix the order and use as a tag system.
But eventually this evolves to something else when I started to deeply using ccxt and created some artificially controlled order flow system outside the exchange, enforcing the use of orders structure.
The same idea was applied for other structures like balance, due to the exchange inconsistency in BALANCE that drive to a huge trading problems when you deal with hedging and arbitrage.

the suggestion was to extend the order structure to incorporate and manage tagging orders.
